tabcmd/commands/server.py

```diff
@@ -61,18 +61,31 @@ def get_items_by_name(logger, item_endpoint, item_name: str, container: Optional
             req_option.filter.add(
                 TSC.Filter(TSC.RequestOptions.Field.Name, TSC.RequestOptions.Operator.Equals, item_name)
             )
+
+            # todo - this doesn't filter if the project is in the top level.
+            # todo: there is no guarantee that these fields are the same for different content types.
+            # probably better if we move that type specific logic out to a wrapper
+            if container:
+                # the name of the filter field is different if you are finding a project or any other item
+                if type(item_endpoint).__name__.find("Projects") < 0:
+                    parentField = TSC.RequestOptions.Field.ProjectName
+                    parentValue = container.name
+                else:
+                    parentField = TSC.RequestOptions.Field.ParentProjectId
+                    parentValue = container.id
+                logger.debug("filtering for parent with {}".format(parentField))
+
+                req_option.filter.add(TSC.Filter(parentField, TSC.RequestOptions.Operator.Equals, parentValue))
+
             all_items, pagination_item = item_endpoint.get(req_option)
 
-            if all_items is None or all_items == []:
+            if pagination_item.total_available == 0:
```

Comment:
Would an error response(4xx or 5xx status code) from the server contain the `pagination_item`?

Reply:
No, all http errors will be raised as an Exception (see tsc endpoint.py::check_status)